code/game/turfs/space/space.dm

```diff
@@ -60,10 +60,10 @@
 /turf/open/space/attack_paw(mob/user)
 	return src.attack_hand(user)
 
-/turf/open/space/attackby(obj/item/C, mob/user, area_restriction)
+/turf/open/space/attackby(obj/item/C, mob/user, area/area_restriction)
 	..()
 	if(istype(C, /obj/item/stack/rods))
-		if(area_restriction && !istype(get_area(src), area_restriction))
+		if(istype(area_restriction) && !istype(get_area(src), area_restriction))
```

Comment:
Shit fix. The third argument to ```attackby()``` is the Click params, this is why things didn't work, move your crap to be the 4th arg.